### PR TITLE
Adds PDF preparer code for race and ethnicity

### PR DESCRIPTION
**Pull request by bseeger**
Note that the race and ethnicity for the applicant are covered via mapping in the pdf-map.yml file. I didn't have to do anything explicitly for those except to tweak the mapping a little. 

The mapping for household member's got tucked into the household details preparer.

### CODE REVIEW

File: src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java [hunk 1]
```diff
@@ -53,9 +53,9 @@ public static ArrayList<LinkedHashMap> formattedHouseholdData(Submission submiss
 
   }
 
-  protected static LinkedHashMap householdData(ArrayList<LinkedHashMap> household, String uuid) {
-    LinkedHashMap user = new LinkedHashMap();
-    for (LinkedHashMap hhmember : household) {
+  private static Map<String, Object> householdData(List<Map<String, Object>> household, String uuid) {
+    Map<String, Object> user = new LinkedHashMap<>();
+    for (Map<String, Object> hhmember : household) {
```
Comment (bseeger) [inline src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java:58]: I don't generally change code outside of the PR, but I had added a method to this class and updated a few other things in the process.  I later took the method out as I didn't really need it. 

I can reverse these if you want, but I think it's good practice to use generic classes versus being so specific.

File: src/main/resources/messages_es.properties [hunk 2]
```diff
@@ -960,11 +960,12 @@ ethnicity-selection.option2=No hispano ni latino
 race-selection.title=Selecci�n de raza
 race-selection.header=�Con qu� razas se identifica?
 race-selection.subheader=Seleccione todas las opciones que correspondan.
-race-selection.option1=Ind�gena americana o nativa de Alaska
-race-selection.option2=Asi�tica
-race-selection.option3=Negra o afroamericana
-race-selection.option4=Nativa de Haw�i o de otras islas del Pac�fico
-race-selection.option5=Cauc�sica
+race-selection.option1=Ind�gena americana
+race-selection.option2=Nativa de Alaska
```
Comment (bseeger) [inline src/main/resources/messages_es.properties:964]: This and the VI translations need to be double checked or overwritten by transifex. Not sure how that process works.

File: src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java [hunk 3]
```diff
@@ -45,6 +50,23 @@ public Map<String, SubmissionField> prepareSubmissionFields(Submission submissio
 
         var uuid = householdMember.get("uuid");
         results.put("householdUSCitizen" + i, new SingleField("householdUSCitizenDerived", nonCitizens != null && nonCitizens.contains(uuid) ? "No" : "Yes", i + 1));
+
+        List<String> raceInput = (List)submission.getInputData().get(RACE_PREFIX + uuid + "[]");
```
Comment (analoo) [inline src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java:54]: @bseeger we have a field called "permissionToAskAboutRace" that  determines whether the race questions were asked. If it's false then we wouldn't have displayed the race and ethnicity questions to the form filler.  

Maybe there is a benefit to using that here to exit out of the process sooner.

File: src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java [hunk 4]
```diff
@@ -45,6 +50,23 @@ public Map<String, SubmissionField> prepareSubmissionFields(Submission submissio
...
+              .collect(Collectors.joining(","));
+        }
+
+        if (ethnicityInput != null && !ethnicityInput.isEmpty()) {
+          raceEthnicCode += "/" + EthnicityType.getAbbreviationFromValue(ethnicityInput);
+        }
+        results.put("householdRaceEthnicCode" + i, new SingleField("householdRaceEthnicCode", raceEthnicCode, i+1));
       }
     }
     return results;
```
Comment (analoo) [inline src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java:72]: This is more for me: curious about whether it makes sense to make each of these inout checks into a method separate from the main method. Just noticing that it's getting pretty big. I could also see a good reason to split it out into methods for testing purposes.

File: src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java (hunk 1 above)
Comment (analoo) [reply src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java:58]: This is great!

File: src/main/resources/pdf-map.yaml [hunk 5]
```diff
@@ -167,6 +168,7 @@ subflowInfo:
       householdHighestEducationFormatted: householdHighestEducation
       householdBirthdayFormatted: householdBirthday
       householdUSCitizenDerived: householdUSCitizen
+      householdRaceEthnicCode: householdRaceEthnicCode
```
Comment (analoo) [inline src/main/resources/pdf-map.yaml:171]: should this be householdRaceEthnicityCode?

File: src/main/resources/messages_es.properties (hunk 2 above)
Comment (sree-cfa) [reply src/main/resources/messages_es.properties:964]: Yeah, once this is merged into main, we can double check it in [transifex](https://app.transifex.com/code-for-america-1/snil-la-doc-upload/translate/#es/src-main-resources-messages-properties--main/475498028?q=key%3Arace)

**Review by sree-cfa — APPROVED** [review]
Also LGTM - thanks for all the cleanup

File: src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java (hunk 3 above)
Comment (bseeger) [reply src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java:54]: Adding in this logic. Thanks for pointing that out, @analoo.

File: src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java (hunk 4 above)
Comment (bseeger) [reply src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java:60]: I did refactor the code a little here.

**Pull request merged**